subsys/bluetooth/controller/ll_sw/ull_llcp_cc.c

```diff
@@ -0,0 +1,603 @@
+
+	pdu->cig_id = ctx->data.cis_create.cig_id;
+	pdu->cis_id = ctx->data.cis_create.cis_id;
+	pdu->cis_handle = ctx->data.cis_create.cis_handle;
+
+	ctx->data.cis_create.host_request_to = 0U;
+
+	/* Enqueue notification towards LL */
+	ll_rx_put(ntf->hdr.link, ntf);
+	ll_rx_sched();
+}
+
+
+static void rp_cc_ntf(struct ll_conn *conn, struct proc_ctx *ctx)
+{
+	struct node_rx_pdu *ntf;
+	struct node_rx_conn_iso_estab *pdu;
+
+	/* Allocate ntf node */
+	ntf = llcp_ntf_alloc();
+	LL_ASSERT(ntf);
+
+	ntf->hdr.type = NODE_RX_TYPE_CIS_ESTABLISHED;
+	ntf->hdr.handle = conn->lll.handle;
+	ntf->hdr.rx_ftr.param = ll_conn_iso_stream_get(ctx->data.cis_create.cis_handle);
+
+	pdu = (struct node_rx_conn_iso_estab *)ntf->pdu;
+
+	pdu->cis_handle = ctx->data.cis_create.cis_handle;
+	pdu->status = ctx->data.cis_create.error;
+
+	/* Enqueue notification towards LL */
+	ll_rx_put(ntf->hdr.link, ntf);
+	ll_rx_sched();
+}
+
+static void rp_cc_complete(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt, void *param)
+{
+	if (!llcp_ntf_alloc_is_available()) {
+		ctx->state = RP_CC_STATE_WAIT_NTF;
+	} else {
+		rp_cc_ntf(conn, ctx);
+		llcp_rr_complete(conn);
+		ctx->state = RP_CC_STATE_IDLE;
+	}
+}
+
+static void rp_cc_send_cis_rsp(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+			       void *param)
+{
+	if (llcp_rr_ispaused(conn) || !llcp_tx_alloc_peek(conn, ctx)) {
+		ctx->state = RP_CC_STATE_WAIT_TX_CIS_RSP;
+	} else {
+		llcp_rp_cc_tx_rsp(conn, ctx);
+
+		/* Wait for the LL_CIS_IND */
+		ctx->rx_opcode = PDU_DATA_LLCTRL_TYPE_CIS_IND;
+		ctx->state = RP_CC_STATE_WAIT_RX_CIS_IND;
+	}
+}
+
+static void rp_cc_send_create_ntf(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				void *param)
+{
+	if (!llcp_ntf_alloc_is_available()) {
+		ctx->state = RP_CC_STATE_WAIT_NTF_CIS_CREATE;
+	} else {
+		rp_cc_ntf_create(conn, ctx);
+		ctx->state = RP_CC_STATE_WAIT_REPLY;
+	}
+}
+
+static void rp_cc_send_reject_ind(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+				  void *param)
+{
+	if (llcp_rr_ispaused(conn) || !llcp_tx_alloc_peek(conn, ctx)) {
+		ctx->state = RP_CC_STATE_WAIT_TX_REJECT_IND;
+	} else {
+		llcp_rp_cc_tx_reject(conn, ctx, PDU_DATA_LLCTRL_TYPE_CIS_REQ);
+
+		if (ctx->data.cis_create.error == BT_HCI_ERR_CONN_ACCEPT_TIMEOUT) {
+			/* We reject due to an accept timeout, so we should generate NTF */
+			rp_cc_complete(conn, ctx, evt, param);
+		} else {
+			/* Otherwise we quietly complete the procedure */
+			llcp_rr_complete(conn);
+			ctx->state = RP_CC_STATE_IDLE;
+		}
+
+	}
+}
+
+static void rp_cc_state_idle(struct ll_conn *conn, struct proc_ctx *ctx, uint8_t evt,
+			     void *param)
+{
+	switch (evt) {
+	case RP_CC_EVT_RUN:
+		ctx->state = RP_CC_STATE_WAIT_RX_CIS_REQ;
+		break;
+	default:
+		/* Ignore other evts */
+		break;
+	}
+}
+
+static inline bool phy_valid(uint8_t phy)
+{
+	/* This is equivalent to:
+	 * exactly one bit set, and no bit set is rfu's
+	 */
+	return (phy == 1 || phy == 2 || phy == 4);
```

Comment:
I'm sure we have some `#define`s for these PHY values somewhere :) Please use those

Reply:
I specifically didn't use the PHY defines because I had seen it as an abstraction, but it is equivalent, so I'll just use the defines to avoid soar eyes ;-)